api/app/jobs/noaa.py

```diff
@@ -0,0 +1,241 @@
+                            if downloaded:
+                                self.files_downloaded += 1
+                                # If we've downloaded the file ok, we can now process it.
+                                try:
+                                    self.grib_processor.process_grib_file(
+                                        downloaded, model_info, session)
+                                    # Flag the file as processed
+                                    flag_file_as_processed(url, session)
+                                    self.files_processed += 1
+                                finally:
+                                    # delete the file when done.
+                                    os.remove(downloaded)
+            except Exception as exception:
+                self.exception_count += 1
+                # We catch and log exceptions, but keep trying to download.
+                # We intentionally catch a broad exception, as we want to try and download as much
+                # as we can.
+                logger.error('unexpected exception processing %s',
+                             url, exc_info=exception)
+
+    def process_model_run(self, model_cycle: str):
+        """ Process a particular model run """
+        logger.info('Processing {} model run cycle {}'.format(
+            self.model_type, model_cycle))
+
+        # Get the urls for the current model run.
+        urls = list(get_gfs_model_run_download_urls(self.now, model_cycle))
+
+        # Process all the urls.
+        self.process_model_run_urls(urls)
+
+        # Having completed processing, check if we're all done.
+        with app.db.database.get_write_session_scope() as session:
+            if check_if_model_run_complete(session, urls):
+                logger.info(
+                    '{} model run {} completed with SUCCESS'.format(self.model_type, model_cycle))
+                prediction_run_timestamp, _ = parse_url_for_timestamps(url=urls[0])
+                mark_prediction_model_run_processed(
+                    session, self.model_type, self.projection, prediction_run_timestamp)
+
+    def process(self):
+        """ Entry point for downloading and processing weather model grib files """
+        for hour in get_gfs_model_run_hours():
+            try:
+                self.process_model_run(hour)
+            except Exception as exception:
+                # We catch and log exceptions, but keep trying to process.
+                # We intentionally catch a broad exception, as we want to try to process as much as we can.
+                self.exception_count += 1
+                logger.error(
+                    'unexpected exception processing %s model run %d',
+                    self.model_type, hour, exc_info=exception)
+
+
+def process_models():
+    """ downloading and processing models """
+
+    # grab the start time.
+    start_time = datetime.datetime.now()
```

Comment:
Does this need to be using a specific timezone or do we just use the default on the pods where this runs?